### PR TITLE
Now KafkaSource supports cloudevents Kafka Structured Encoding

### LINKED ISSUE #842: How to deal with embedded CloudEvent message ?
What if a source is already reading CloudEvents :slightly_smiling_face: ?

Let's assume that that on an Apache Kafka topic we might have some data as CloudEvent already (for what ever reason). 

Should we ACK that, and *not* wrap out CloudEvent data around the received CloudEvent ?

Currently this would like:

```
☁️  cloudevents.Event
Validation: valid
Context Attributes,
  specversion: 1.0
  type: dev.knative.kafka.event
  source: /apis/v1/namespaces/default/kafkasources/kafka-source-simple#temperature-values-ce
  subject: partion:0#1085
  id: partition:0/offset:1085
  time: 2020-01-22T10:36:28.171Z
  datacontenttype: application/json
Extensions,
  key: 7
Data,
  {
    "data": {
      "stationId": 7,
      "timestamp": "2020-01-22T10:36:28.170Z",
      "value": -33.7
    },
    "id": "20dee11e-e112-4e1c-9948-7a088e541448",
    "source": "/blah/something/temperature-values",
    "specversion": "0.3",
    "type": "blah.something.measurement"
  }
```

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Following spec: https://github.com/cloudevents/spec/blob/v1.0/kafka-protocol-binding.md#33-structured-content-mode

Also fixed bad extension name when dumping kafka headers to event extensions

Fixes #842

### CODE REVIEW

File: test/e2e-tests.sh [hunk 1]
```diff
@@ -172,7 +172,7 @@ function kafka_setup() {
   sed 's/namespace: .*/namespace: kafka/' ${STRIMZI_INSTALLATION_CONFIG_TEMPLATE} > ${STRIMZI_INSTALLATION_CONFIG}
   kubectl apply -f "${STRIMZI_INSTALLATION_CONFIG}" -n kafka
   kubectl apply -f ${KAFKA_INSTALLATION_CONFIG} -n kafka
-  kubectl apply -f ${KAFKA_TOPIC_INSTALLATION_CONFIG} -n kafka
+  # kubectl apply -f ${KAFKA_TOPIC_INSTALLATION_CONFIG} -n kafka
```
Comment (lberk) [inline test/e2e-tests.sh]: left as a comment?

File: test/e2e/helpers/kafka_helper.go [hunk 2]
```diff
@@ -21,10 +21,22 @@ import (
...
 )
 
+const (
+	strimziApiGroup   = "kafka.strimzi.io"
+	strimziApiVersion = "v1beta1"
+	strimziKafkaTopic = "KafkaTopic"
+)
+
+var (
+	topicGVR = schema.GroupVersionResource{Group: strimziApiGroup, Version: strimziApiVersion, Resource: "kafkatopics"}
```
Comment (lberk) [inline test/e2e/helpers/kafka_helper.go]: use of two constants and resource as a string?

File: test/e2e/helpers/kafka_helper.go [hunk 3]
```diff
@@ -94,3 +106,35 @@ func MustPublishKafkaMessage(client *testlib.Client, bootstrapServer string, top
 		client.T.Fatalf("Failed waiting for pod for completeness %q: %v", pod.Name, err)
 	}
 }
+
+func MustCreateTopic(client *testlib.Client, clusterName string, clusterNamespace string, topicName string) {
+	obj := unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": strimziApiGroup + "/" + strimziApiVersion,
+			"kind":       "KafkaTopic",
```
Comment (lberk) [inline test/e2e/helpers/kafka_helper.go]: didn't you define this as a const?

File: test/e2e-tests.sh (hunk 1 above)
Comment (slinkydeveloper) [reply test/e2e-tests.sh]: It's deliberate: it's not needed anymore but i wish to leave it for anybody who needs a quick test topic

Comment (lberk) [reply test/e2e-tests.sh]: I'm not sure somebody will be looking at knative e2e testing for basic strimzi configuration setup?

**Comment by grantr** [comment]
Release notes followup: This was a new feature and should have had a release note blurb :smiley:

**Comment by slinkydeveloper** [comment]
@grantr I'm sorry, should i add it directly here in the PR description?

**Comment by grantr** [comment]
@slinkydeveloper for future PRs, use the release note block in the PR template when making a user-visible change (the template has instructions). If everyone does that, we can auto-generate the release notes instead of waiting for a human to spend hours copying and pasting :)

**Pull request merged**